### PR TITLE
SFX-300 - Update stories in Storybook to be presentable

### PR DESCRIPTION
**Pull request by ChantalDesrochers**
- Removed all 'setTimeouts' and replaced with Julia's custom events addon
- Updated notes portion to be consistent across all stories
- Removed stories made unnecessary with the addition of the custom events addon
- Removed unnecessary buttons/functions in the components themselves
- Changed method to generate html and knobs in stories to be consistent across all stories
- Made additions to the SAYT README
- NOTE 1: code formatter made formatting changes to files changed. Please let me know if I should revert - assumption is that linter will eventually standardize and the issue of different style preferences will be resolved
- NOTE 2: SAYT code snippets are pushed to the bottom of the story - this is done to mimic the actual style behaviour of SAYT (SAYT will appear on top of other images).
-- pushed SAYT down so that code snippet is still visible - please let me know if this is desired

### CODE REVIEW

File: packages/web-components/@sfx/sayt/stories/index.ts [hunk 1]
```diff
@@ -150,84 +184,60 @@ storiesOf('Components|SAYT', module)
...
+                  2. Click on one of the close buttons.
+                    * View both SAYT components closing.
+
+      ### Example of the multple SAYT components:
+
+      \`\`\`html
+      <input type="text" id="search-box1" placeholder="Search here" />
+      <sfx-sayt
+        searchbox="search-box1"
+        closetext="x"
```
Comment (jmbtutor) [inline packages/web-components/@sfx/sayt/stories/index.ts]:
```suggestion
        closetext="×"
```

File: packages/web-components/@sfx/sayt/stories/index.ts [hunk 2]
```diff
@@ -150,84 +184,60 @@ storiesOf('Components|SAYT', module)
...
+        showclosebutton
+        visible
+      ></sfx-sayt>
+      <input type="text" id="search-box2" placeholder="Search here" />
+      <sfx-sayt
+        searchbox="search-box2"
+        closetext="x"
```
Comment (jmbtutor) [inline packages/web-components/@sfx/sayt/stories/index.ts]:
```suggestion
        closetext="×"
```

File: packages/web-components/@sfx/search-box/stories/index.ts [hunk 3]
```diff
@@ -1,68 +1,78 @@
...
-  console.log('search box cleared event received');
-  console.log('event', e);
-});
+const searchboxNotesMarkdownIntro = ` # SF-X Search Box Component
 
-window.addEventListener(SEARCHBOX_EVENT.SEARCHBOX_CLICK, e => {
-  console.log('search box clicked event received');
-  console.log('event', e);
-});
+[SF-X Search Box README](https://github.com/groupby/sfx-view/tree/master/packages/web-components/%40sfx/search-box "SF-X Search Box README").
```
Comment (jmbtutor) [inline packages/web-components/@sfx/search-box/stories/index.ts]: Same as similar lines.

File: packages/web-components/@sfx/search-box/stories/index.ts [hunk 4]
```diff
@@ -1,68 +1,78 @@
...
+    name: SEARCHBOX_EVENT.UPDATE_SEARCH_TERM,
+    payload: 'hot chocolate'
+  }
+];
 
-window.addEventListener(SEARCHBOX_EVENT.SEARCHBOX_CLEAR_CLICK, e => {
-  console.log('search box cleared event received');
-  console.log('event', e);
-});
+const searchboxNotesMarkdownIntro = ` # SF-X Search Box Component
```
Comment (jmbtutor) [inline packages/web-components/@sfx/search-box/stories/index.ts]:
```suggestion
const searchboxNotesMarkdownIntro = `# SF-X Search Box Component
```

File: packages/web-components/@sfx/search-box/stories/index.ts [hunk 5]
```diff
@@ -1,68 +1,78 @@
...
+[SF-X Search Box README](https://github.com/groupby/sfx-view/tree/master/packages/web-components/%40sfx/search-box "SF-X Search Box README").
 
-window.addEventListener(SEARCHBOX_EVENT.SEARCHBOX_CHANGE, e => {
-  console.log('search box change event received');
-  console.log('event', e);
-});
-// end of event dispatch testing
+\`\`\`html
+<sfx-search-box
+placeholder="Search Here"
```
Comment (jmbtutor) [inline packages/web-components/@sfx/search-box/stories/index.ts]: I think we want these indented.

File: packages/web-components/@sfx/search-box/stories/index.ts [hunk 6]
```diff
@@ -1,68 +1,78 @@
...
+searchbutton
+clearbutton
+></sfx-search-box>
+\`\`\`
+
+## Demonstrated in this story:`;
```
Comment (jmbtutor) [inline packages/web-components/@sfx/search-box/stories/index.ts]:
```suggestion
## Demonstrated in this story`;
```

File: packages/web-components/@sfx/search-box/stories/index.ts [hunk 7]
```diff
@@ -1,68 +1,78 @@
...
-    )}"></sfx-search-box>
-    <button @click="${this.updateTextEvent}">
-      Click to dispatch update search term event
-    </button>
-  `
-    );
+  .add(
+    'Rendering based on events and attributes',
+    () => {
+      const searchBoxComponent = getSearchBoxComponent();
```
Comment (jmbtutor) [inline packages/web-components/@sfx/search-box/stories/index.ts]: Is there a point to this `const` if we're only using it once and we're also doing a function call three lines down?

File: packages/web-components/@sfx/search-box/stories/index.ts [hunk 8]
```diff
@@ -1,68 +1,78 @@
...
+      return `
+      ${searchBoxComponent}
+      ${getDisplayCode(searchBoxComponent)}
+      `;
+    },
+    {
+      customEvents: updateTextEvent,
+      notes: {
+        markdown: `
+        ${searchboxNotesMarkdownIntro}
```
Comment (jmbtutor) [inline packages/web-components/@sfx/search-box/stories/index.ts]: Indentation.

File: packages/web-components/@sfx/search-box/stories/index.ts [hunk 9]
```diff
@@ -1,68 +1,78 @@
...
+
+          * The SF-X Search Box component rendering with customizations.
```
Comment (jmbtutor) [inline packages/web-components/@sfx/search-box/stories/index.ts]: Rather than this, I would reword and promote the subpoints as top-level points.

```suggestion
          * The SF-X Search Box component renders with customizations.
```

File: packages/web-components/@sfx/search-box/stories/index.ts [hunk 10]
```diff
@@ -1,68 +1,78 @@
...
+            * Clear and search buttons are optional.
```
Comment (jmbtutor) [inline packages/web-components/@sfx/search-box/stories/index.ts]: With the suggestion above, maybe something like `The SF-X Search Box component can render "clear" and "search" buttons.`

File: packages/web-components/@sfx/search-box/stories/index.ts [hunk 11]
```diff
@@ -1,68 +1,78 @@
...
+              * If the 'clearbutton' attribute is present, a clear button will display.
```
Comment (jmbtutor) [inline packages/web-components/@sfx/search-box/stories/index.ts]: As a reminder, attribute names should be in backticks. This continues to apply to the rest of the PR.

```suggestion
              * If the \`clearbutton\` attribute is present, a clear button will display.
```

File: packages/web-components/@sfx/search-box/stories/index.ts [hunk 12]
```diff
@@ -1,68 +1,78 @@
...
+              * If the 'searchbutton' attribute is present, a search button will display.
+                * To add and remove these attributes:
+                  1. Visit the **Knobs** tab and toggle the 'Show search button' and 'Show clear button'.
```
Comment (jmbtutor) [inline packages/web-components/@sfx/search-box/stories/index.ts]: And another reminder to use double quotes in English text.

```suggestion
                  1. Visit the **Knobs** tab and toggle the "Show search button" and "Show clear button".
```

File: packages/web-components/@sfx/search-box/stories/index.ts [hunk 13]
```diff
@@ -1,68 +1,78 @@
...
+                    * View the DOM and component update accordingly.
```
Comment (jmbtutor) [inline packages/web-components/@sfx/search-box/stories/index.ts]: See other similar lines.

File: packages/web-components/@sfx/search-box/stories/index.ts [hunk 14]
```diff
@@ -1,68 +1,78 @@
...
+            * The placeholder text can be customized.
+                * If the 'placeholder' attribute is present with a string value, the placeholder text will replace the default placeholder (Type your search)
```
Comment (jmbtutor) [inline packages/web-components/@sfx/search-box/stories/index.ts]: I think there's one level too much indentation space-wise.

Perhaps I would word this as
```
The \`placeholder\` attribute defines the placeholder text of the search box. It defaults to "Type your search" if not present.
```

File: packages/web-components/@sfx/search-box/stories/index.ts [hunk 15]
```diff
@@ -1,68 +1,78 @@
...
+                  * To modify the 'placeholder' attribute:
+                    1. Visit the **Knobs** tab and modify the text contained within the 'Placeholder Text' field.
+                      * View the component and DOM update accordingly.
```
Comment (jmbtutor) [inline packages/web-components/@sfx/search-box/stories/index.ts]: Same as similar lines.

File: packages/web-components/@sfx/search-box/stories/index.ts [hunk 16]
```diff
@@ -1,68 +1,78 @@
...
+          * The SF-X Search Box component populating with an updated search term in response to the \`sfx::update_search_term\` event.
```
Comment (jmbtutor) [inline packages/web-components/@sfx/search-box/stories/index.ts]: Also use the event constant.

```suggestion
          * The SF-X Search Box component populates with an updated search term in response to the \`sfx::update_search_term\` event.
```

File: packages/web-components/@sfx/search-box/stories/index.ts [hunk 17]
```diff
@@ -1,68 +1,78 @@
...
+              * To emit the event:
+                1. Navigate to the **Custom Events** tab and locate the \`sfx::update_search_term\` event.
+                2. Click 'emit'.
+                  * View the search term populate in the search box input.
```
Comment (jmbtutor) [inline packages/web-components/@sfx/search-box/stories/index.ts]: Same as similar lines.

File: packages/web-components/@sfx/search-box/stories/index.ts [hunk 18]
```diff
@@ -1,68 +1,78 @@
...
+                  1. Visit the **Knobs** tab and toggle the 'Show search button' and 'Show clear button'.
+                    * View the DOM and component update accordingly.
+            * The placeholder text can be customized.
+                * If the 'placeholder' attribute is present with a string value, the placeholder text will replace the default placeholder (Type your search)
+                  * To modify the 'placeholder' attribute:
+                    1. Visit the **Knobs** tab and modify the text contained within the 'Placeholder Text' field.
+                      * View the component and DOM update accordingly.
+          * The SF-X Search Box component populating with an updated search term in response to the \`sfx::update_search_term\` event.
+              * To emit the event:
+                1. Navigate to the **Custom Events** tab and locate the \`sfx::update_search_term\` event.
```
Comment (jmbtutor) [inline packages/web-components/@sfx/search-box/stories/index.ts]: Use the event constant here.

File: packages/web-components/@sfx/ui/stories/index.ts [hunk 19]
```diff
@@ -17,19 +18,51 @@ const items = [
...
+      return `
+      ${getListComponent()}
+      ${getDisplayCode(getListComponent(true))}
+    `;
+    },
+    {
+      notes: {
+        markdown: `# SF-X List Component
+
+        [SF-X UI README](https://github.com/groupby/sfx-view/tree/master/packages/web-components/%40sfx/ui "SF-X List README")
```
Comment (jmbtutor) [inline packages/web-components/@sfx/ui/stories/index.ts]: Same as similar lines.

File: packages/web-components/@sfx/ui/stories/index.ts [hunk 20]
```diff
@@ -17,19 +18,51 @@ const items = [
...
+
+        \`\`\`html
+        <sfx-ui
+          caption="Items"
+          list=[]
```
Comment (jmbtutor) [inline packages/web-components/@sfx/ui/stories/index.ts]: The attribute value should be quoted. I think this should also be `items`.

```suggestion
          items="[]"
```

File: packages/web-components/@sfx/ui/stories/index.ts [hunk 21]
```diff
@@ -17,19 +18,51 @@ const items = [
...
+    `;
+    },
+    {
+      notes: {
+        markdown: `# SF-X List Component
+
+        [SF-X UI README](https://github.com/groupby/sfx-view/tree/master/packages/web-components/%40sfx/ui "SF-X List README")
+
+        \`\`\`html
+        <sfx-ui
```
Comment (jmbtutor) [inline packages/web-components/@sfx/ui/stories/index.ts]: The tag name is incorrect.

```suggestion
        <sfx-list
```

File: packages/web-components/@sfx/ui/stories/index.ts [hunk 22]
```diff
@@ -17,19 +18,51 @@ const items = [
...
+          caption="Items"
+          list=[]
+        ></sfx-ui>
+        \`\`\`
+
+        ## Demonstrated in this story:
```
Comment (jmbtutor) [inline packages/web-components/@sfx/ui/stories/index.ts]:
```suggestion
        ## Demonstrated in this story
```

File: packages/web-components/@sfx/ui/stories/index.ts [hunk 23]
```diff
@@ -17,19 +18,51 @@ const items = [
...
+
+          * The SF-X List component rendering a list with data populated via the 'caption' and 'list' attributes.
```
Comment (jmbtutor) [inline packages/web-components/@sfx/ui/stories/index.ts]: Again, I think the subpoints should reworded and promoted, but to fix this line as-is: 

```suggestion
          * The SF-X List component renders a list with data from the \`caption\` and \`list\` attributes.
```

File: packages/web-components/@sfx/ui/stories/index.ts [hunk 24]
```diff
@@ -17,19 +18,51 @@ const items = [
...
+            * To modify the 'list' attribute data (the list items):
```
Comment (jmbtutor) [inline packages/web-components/@sfx/ui/stories/index.ts]: If this is promoted, I would word this as
```
The \`list\` attribute defines the list items.
```

File: packages/web-components/@sfx/ui/stories/index.ts [hunk 25]
```diff
@@ -17,19 +18,51 @@ const items = [
...
+              1. Visit the **Knobs** tab.
+              2. Modify the data inside the 'List Items' field.
+                * View the component update with the updated data.
+            * To modify the 'caption' attribute text (the optional title):
```
Comment (jmbtutor) [inline packages/web-components/@sfx/ui/stories/index.ts]: If this is promoted, I would word this as
```
The `caption` attribute defines the list title text.
```

File: packages/web-components/@sfx/ui/stories/index.ts [hunk 26]
```diff
@@ -17,19 +18,51 @@ const items = [
...
+              1. Visit the **Knobs** tab.
+              2. Update the text in the in the SF-X 'List Title' field.
+                * View the DOM and the component update with the updated text.
+          * NOTE: The SF-X List component is intended to be used as an UI component in conjunction with other components (e.g. the SF-X Autocomplete component utilizes the SF-X List component).
```
Comment (jmbtutor) [inline packages/web-components/@sfx/ui/stories/index.ts]: I think this should go closer to the top, perhaps even before this section.

```suggestion
          * NOTE: The SF-X List component is intended to be used as a "building block" UI component inside other components (e.g. the SF-X Autocomplete component uses the SF-X List component as part of its template).
```

File: packages/web-components/@sfx/autocomplete/stories/index.ts [hunk 27]
```diff
@@ -1,100 +1,51 @@
...
-/*
- * --- Slots Testing ---
- * The final story demonstrates that content slotted in (in this case,
- * the <div> contained within the autocomplete custom element)
- * appears within the light-dom as desired.
- *
- */
+function getAutocomplete(): string {
+  const optionalTitle = text('Optional Caption', 'Autocomplete Results');
+  return '<sfx-autocomplete\n' + ` caption="${optionalTitle}"\n` + '></sfx-autocomplete>';
```
[the remark this answers is not in the document]
Comment (ChantalDesrochers) [reply packages/web-components/@sfx/autocomplete/stories/index.ts]: I had it on separate lines but my formatter moved to one. I think it looks better on separate lines so I will override my formatter.
